src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -1,152 +1,119 @@
+            public (IOperation Condition, bool JumpIfTrue, Branch Branch) InternalConditional;
+            public (IOperation Value, Branch Branch) InternalNext;
+            
+            public BasicBlock(BasicBlockKind kind)
+            {
+                Kind = kind;
+                _statementsBuilder = ArrayBuilder<IOperation>.GetInstance();
+                _predecessorsBuilder = new HashSet<BasicBlock>();
+            }
 
-        public int Ordinal { get; internal set; } = -1;
+            public IReadOnlyCollection<IOperation> Statements => _statementsBuilder;
+            public ImmutableArray<IOperation> GetImmutableStatements() => _statementsBuilder.ToImmutableAndFree();
+            public ISet<BasicBlock> Predecessors => _predecessorsBuilder;
+            public BasicBlockKind Kind { get; }
 
-        public bool IsReachable { get; internal set; } = false;
+            public int Ordinal { get; set; } = -1;
 
-        /// <summary>
-        /// Enclosing region
-        /// </summary>
-        public ControlFlowGraph.Region Region { get; internal set; }
+            public bool IsReachable { get; set; } = false;
 
-        internal void AddStatement(IOperation statement)
-        {
-            _statements.Add(statement);
-        }
+            public ControlFlowRegion Region { get; set; }
 
-        internal void AddStatements(ImmutableArray<IOperation> statements)
-        {
-            _statements.AddRange(statements);
-        }
+            internal void AddStatement(IOperation statement)
+            {
+                _statementsBuilder.Add(statement);
+            }
 
-        internal void RemoveStatements()
-        {
-            _statements.Clear();
-        }
+            internal void AddStatements(IEnumerable<IOperation> statements)
+            {
+                _statementsBuilder.AddRange(statements);
+            }
 
-        internal void AddPredecessor(BasicBlock block)
-        {
-            _predecessors.Add(block);
-        }
+            internal void RemoveStatements()
+            {
+                _statementsBuilder.Clear();
+            }
 
-        internal void RemovePredecessor(BasicBlock block)
-        {
-            _predecessors.Remove(block);
-        }
+            internal void AddPredecessor(BasicBlock block)
+            {
+                _predecessorsBuilder.Add(block);
+            }
 
-        public enum BranchKind
-        {
-            None,
-            Regular,
-            Return,
-            StructuredExceptionHandling,
-            ProgramTermination,
-            Throw,
-            ReThrow,
-            Error,
-        }
+            internal void RemovePredecessor(BasicBlock block)
+            {
+                _predecessorsBuilder.Remove(block);
+            }
 
-        public struct Branch
-        {
-            private ImmutableArray<ControlFlowGraph.Region> _lazyFinallyRegions;
+            public struct Branch
```

Comment:
Should be internal?

Reply:
It is nested within `BasicBlockBuilder`.

Reply:
Yes, but BasicBlock was changed from public to internal, so this should be too for consistency.

---
In reply to: [192250068](https://github.com/dotnet/roslyn/pull/27254#discussion_r192250068) [](ancestors = 192250068)